packages/nextra-theme-docs/src/components/highlight-matches.tsx

```diff
@@ -16,26 +16,27 @@ export const HighlightMatches = memo<MatchArgs>(function HighlightMatches({
   }
   const splitText = value.split('')
   const escapedSearch = escapeStringRegexp(match.trim())
-  const regexp = new RegExp(escapedSearch.replaceAll(' ', '|'), 'ig')
+  const regexp = new RegExp(escapedSearch.replaceAll(/\s+/g, '|'), 'ig')
   let result
   let index = 0
   const content: (string | ReactNode)[] = []
 
-  while (
-    (result = regexp.exec(value)) &&
-    // case `>  ` replaced previously to `>||` + some character provoke memory leak because
-    // `RegExp#exec` will always return a match
-    regexp.lastIndex !== 0
-  ) {
-    const before = splitText.splice(0, result.index - index).join('')
-    const after = splitText.splice(0, regexp.lastIndex - result.index).join('')
-    content.push(
-      before,
-      <span key={result.index} className="nx-text-primary-600">
-        {after}
-      </span>
-    )
-    index = regexp.lastIndex
+  while ((result = regexp.exec(value))) {
+    if (result.index === regexp.lastIndex) {
+      regexp.lastIndex++
+    } else {
```

Comment:
just curious, behaviour should be the same as previously?
`result.index === regexp.lastIndex` only when `index = 0`?
So you increment `lastIndex` to exit loop?

Reply:
The `result.index === regexp.lastIndex` condition in the `highlight-matches` component serves as a safeguard against infinite loops that can occur when the regular expression matches an empty string. Here's a simple example to illustrate this:

```js
const value = 'Docs Theme';
const match = '  ';
const escapedSearch = escapeStringRegexp(match.trim());
const regexp = new RegExp(escapedSearch.replaceAll(/\s+/g, '|'), 'ig'); // -> /(?:)/gi
let result;

while ((result = regexp.exec(value))) {
  if (result.index === regexp.lastIndex) {
    // lastIndex: 0, 1, 2...10.  (value.length === 10)
    // This condition is true when the regex matches an empty string.
    // Without incrementing lastIndex, the loop would be infinite.
    regexp.lastIndex++;
  } else {
    // Process the match...
  }
}
```

In this example, the regex `/(?:)/gi` matches an empty string, and without incrementing `regexp.lastIndex`, the `exec` method would keep matching at the same position, causing an infinite loop.
However, the condition to increment `lastIndex` is not currently necessary in the`highlight-matches` component. This is because, in the `flexsearch` component that utilizes `highlight-matches`, if `search` is an empty or whitespace-only string, `highlight-matches` is not invoked. Therefore, the `match` in `highlight-matches` will never be assigned a value that could trigger an infinite loop, and the condition can be safely removed.
Nonetheless, the `result.index === regexp.lastIndex` condition was included as a precaution to ensure that `highlight-matches` can handle all inputs without falling into an infinite loop, even in scenarios where it is not used within `flexsearch` component.